python/tests/simplify.py

```diff
@@ -432,8 +433,57 @@ def finalise_references(self):
         # We don't support migrations for now. We'll need to remap these as well.
         assert self.ts.num_migrations == 0
 
+    def add_old_roots(self):
+        # This is just a prototype, so I'm being lazy and using an
+        # interval library.
+        import portion
+
+        for input_id, output_id in enumerate(self.node_id_map):
+            self.used_up_ancestry[input_id].sort()
+            x = self.A_head[input_id]
+            # Find any gaps in the used-up ancestry
+            gaps = []
+            last_right = 0
+            for left, right in self.used_up_ancestry[input_id]:
+                if left != last_right:
+                    gaps.append(portion.openclosed(last_right, left))
+                last_right = right
+            if last_right != self.tables.sequence_length:
+                gaps.append(portion.openclosed(last_right, self.tables.sequence_length))
+
+            # print(gaps)
+            ancestry = portion.IntervalDict()
+            while x is not None:
+                ancestry[portion.openclosed(x.left, x.right)] = x.node
+                x = x.next
+
+            num_intersecting = 0
+            for gap in gaps:
+                if len(ancestry[gap]) > 0:
+                    num_intersecting += 1
+                # num_intersecting += 1
+                # print("\t", ancestry[gap])
+            if num_intersecting > 0:
+                # print("ROOT")
+                assert self.node_id_map[input_id] == -1
+                output_id = self.record_node(input_id)
+                # print(input_id, ancestry, gaps)
+                for gap in gaps:
+                    num_intersecting += 1
```

Comment:
Is this count of intersections being used anywhere beyond line 466? It looks like as soon as this for loop is exited, you'll flush edges and then return to the start of the node loop in line 441, when `num_intersecting` will be rest to 0 again.